Update unsigned suffixes

pfl opened the pull request:
Adding more of the 'U' suffixes to unsigned variable values all over the code base except /ext. Part of #9895.

pfl committed:
scripts/coccinelle: Update Coccinelle script for unsigned values

Update the script to detect and update more instances of unsigned
variable assignments when using all four simple rules of arithmetics.

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>

pfl committed:
all: Add 'U' suffix when using unsigned variables

Add a 'U' suffix to values when computing and comparing against
unsigned variables.

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>

pfl commented:
Updated somebody else's code that caused errors.

mike-scott reviewed (approved):
LGTM for LwM2M changes.  @pfl Will these now be caught by PR review?

aescolar reviewed (approved):
+1 for native_posix, nrf52_bsim and their drivers' changes
Have not reviewed the other 550 files ;)

himanshujha199640 reviewed (dismissed):
Cc  @JuliaLawall

@pfl 
Could you please check this once:
```
himanshu@himanshu-Vostro-3559:~/zephyr$ spatch -D report scripts/coccinelle/unsigned_suffix.cocci drivers/pwm/pwm_imx.c
init_defs_builtins: /usr/local/bin/../lib/coccinelle/standard.h
HANDLING: drivers/pwm/pwm_imx.c
drivers/pwm/pwm_imx.c:74:7-10: WARNING: Unsigned 'U' suffix missing
himanshu@himanshu-Vostro-3559:~/zephyr$ spatch -D patch scripts/coccinelle/unsigned_suffix.cocci drivers/pwm/pwm_imx.c
init_defs_builtins: /usr/local/bin/../lib/coccinelle/standard.h
HANDLING: drivers/pwm/pwm_imx.c
diff = 
--- drivers/pwm/pwm_imx.c
+++ /tmp/cocci-output-15854-f2f185-pwm_imx.c
@@ -71,7 +71,7 @@ static int imx_pwm_pin_set(struct device
 
 	LOG_DBG("enabled=%d, pulse_cycles=%d, period_cycles=%d,"
 		    " duty_cycle=%d\n", enabled, pulse_cycles, period_cycles,
-		    (100 * pulse_cycles / period_cycles));
+		    (pulse_cycles * 100U / period_cycles));
 
 	/*
 	 * i.MX PWMv2 has a 4-word sample FIFO.
```
The `patch` case didn't workout as multiplication didn't match.
So, I added the following diff on top of your patch to catch it:
```
diff --git a/scripts/coccinelle/unsigned_suffix.cocci b/scripts/coccinelle/unsigned_suffix.cocci
index f63e2883f4..cb48d7b102 100644
--- a/scripts/coccinelle/unsigned_suffix.cocci
+++ b/scripts/coccinelle/unsigned_suffix.cocci
@@ -100,9 +100,8 @@ identifier r_rewrite.z;
 - C
 + z
 |
-  v *
-- C
-+ z
+- v * C
++ v * z
 |
   v /
 - C
```

erwango commented on drivers/flash/flash_stm32f0x.c:
```diff
@@ -24,7 +24,7 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 			     bool write)
 {
-	return (!write || (offset % 2 == 0 && len % 2 == 0)) &&
+	return (!write || (offset % 2 == 0 && len % 2 == 0U)) &&
```
Missed a value here ?
```
(offset % 2 == 0U && len % 2 == 0U)
```

jhedberg replied:
off_t is signed, AFAIK

erwango replied:
ok, right

erwango reviewed (dismissed):
Did some mess this morning, try to set things back.

erwango commented:
@kgugala, @MaureenHelm, @rveerama1, @Mani-Sadhasivam, @jfischer-phytec-iot, 
@sjanc, @carlescufi, @galak, @nordic-krch, @anangl, @nvlsianpu, @loicpoulain,
@sathishkuttan, @rgundi , @ramakrishnapallala, @jarz-nordic , @thoh-ot , @pfl,
@dcpleung , @gnuless and @nategraff-sif:
You were initially in review of this PR. I've done some mess that removed you somehow.
Now, I can't add you manually (github does not allow to add more than 15 reviewers manually, it appears). So please consider you are still asked for review.
Sorry for the mess.

Mani-Sadhasivam reviewed (approved):
+1 for LED changes

vanwinkeljan reviewed (approved):
display related changes lgtm

pfl committed:
scripts/coccinelle: Update unsigned suffix script

Update unsigned suffix script to properly match multiplication.
Fix provided by Himanshu Jha.

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>

pfl committed:
all: Update unsigend 'U' suffix due to multiplication

As the multiplication rule is updated, new unsigned suffixes
are added in the code.

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>

pfl commented:
> So, I added the following diff on top of your patch to catch it:
Added your changes into a new commit and the updated files into another in order to make the additional changes to stand out.
